### PR TITLE
Replaced boost with c++11 features

### PR DESCRIPTION
**Pull request by mensinda**
Resolves Issues:
  - #409
  - #825
  - #826 (used `std::to_string()`)
  - #827
  - #828 (used `std::unique_ptr<T[]>`)
  - #829

I did not remove the BoostWorkaround yet.

### CODE REVIEW

**Comment by kimkulling** [comment]
Whow, looks great. Thank you very much and a lot fo kudo's for your great work!

**Comment by mensinda** [comment]
No problem. But I have no Idea why the Windows builds are failing.

**Comment by turol** [comment]
Windows is case insensitive and confuses your Math.h with builtin math.h

**Comment by mensinda** [comment]
Thanks. Renamed Math.h

**Comment by kimkulling** [comment]
Assimp-Viewer seems to be not updated correctly.

**Comment by mensinda** [comment]
The build for Visual studio 2010 fails because it does not support the needed c++11 features.

https://msdn.microsoft.com/en-us/library/hh567368.aspx.

**Comment by jdduke** [comment]
For the failed compilation here, Visual Studio 2010 lacks support for range-based for, and its to_string implementation is incomplete.

File: code/STEPFile.h [hunk 1]
```diff
@@ -597,6 +596,9 @@ namespace STEP {
         LazyObject(DB& db, uint64_t id, uint64_t line, const char* type,const char* args);
         ~LazyObject();
 
+        LazyObject( LazyObject const& ) = delete;
```
Comment (turol) [inline code/STEPFile.h]: Delete the move constructor/assign operator too. Or implement them.

Comment (mensinda) [inline code/STEPFile.h]: Done

**Comment by kimkulling** [comment]
Let's merge it, 2010 is disabled.

**Comment by sherief** [comment]
@mensinda Thanks, great work!

**Pull request merged**